examples/model_selection/plot_gp_search.py

```diff
@@ -0,0 +1,69 @@
+"""
+==================================================================
+Plotting Performance of different hyperparameter selection methods
+==================================================================
+
+In this plot you can see the validation scores of a Ridge regression model
+along steps of different hyperparameter selection methods.
+"""
+print(__doc__)
+
+import matplotlib.pyplot as plt
+
+import numpy as np
+from scipy import stats
+
+from sklearn.model_selection import SequentialSearchCV
+from sklearn.model_selection import GridSearchCV
+from sklearn.model_selection import RandomizedSearchCV
+from sklearn.linear_model import Ridge
+from sklearn.datasets import make_regression
+
+# Loading the Digits dataset
+X, y = make_regression(
+    n_samples=50, n_features=500, noise=1., random_state=0)
+
+# build a classifier
+clf = Ridge()
+
+# specify parameters and distributions to sample from
+expon = stats.expon(scale=20)
+expon.random_state = 0  # for reproductible results
+params = {'alpha': expon}
+
+# run randomized search
+gp_search = SequentialSearchCV(
+  clf, params, verbose=1, n_iter=20)
+gp_search.fit(X, y)
+
+# retrieve the maximum score at each iteration
+gp_cum_scores = [np.max(
```

Comment:
it might be easier to read this way, but that might be personal preference

    gp_cum_scores = np.zeros(20)
    gp_cum_scores[0] = gp_search.grid_scores_[0].mean_validation_score
    for i in range(1, 20):
        gp_cum_scores[i] = max(gp_cum_scores[i-1], gp_search.grid_scores_[i].mean_validation_score
)

btw should this and others read gp_cum_best_scores?